### PR TITLE
NotificationEntry: Fall back to title and app name when missing body

### LINKED ISSUE #219: Ensure notification always have a title and body
## Problem

If a notification is sent with only a title, it's a bit off balance:

Screenshot from 2021-08-19 12 32 19

## Proposal

If the notification only contains a title, we should use the title as body and the app name as title

## Prior Art

This is what we do in the server

### PR DESCRIPTION
**Pull request by lorencinibrdev**
Should fix #219.

### CODE REVIEW

**Comment by lorencinibrdev** [comment]
Attaching screenshot:

image

File: src/Widgets/NotificationEntry.vala [hunk 1]
```diff
@@ -94,8 +94,14 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
                 image_overlay.add_overlay (badge_image);
             }
         }
+        
+        var entry_title = notification.summary;
+
```
Comment (cassidyjames) [inline src/Widgets/NotificationEntry.vala:99]: Whitespace is failing lint

```suggestion

        var entry_title = notification.summary;

```

File: src/Widgets/NotificationEntry.vala [hunk 2]
```diff
@@ -145,34 +151,37 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
...
-                string[] lines = body.split ("\n");
-                string stripped_body = lines[0] + "\n";
-                for (int i = 1; i < lines.length; i++) {
-                    stripped_body += lines[i].strip () + " ";
-                }
+        
+        if (entry_body == "") {
+            entry_body = notification.summary;
+        }
+
```
Comment (cassidyjames) [inline src/Widgets/NotificationEntry.vala:158]: Whitespace lint here as well

```suggestion

        if (entry_body == "") {
            entry_body = notification.summary;
        }

```

**Review by cassidyjames — DISMISSED** [review]
Couple of whitespace issues

**Comment by lorencinibrdev** [comment]
Thanks again for the review @cassidyjames. I'm sorry for the trouble.

Is there any way to have these checks tests done before doing a PR?

**Review by danrabbit — CHANGES_REQUESTED** [review]
One more small thing! It looks like `app_name` is guaranteed to be set. If we get valid AppInfo in `notification` we should make sure to set `app_name` there

And yes you could install an run vala-lint ahead of time if you like :)

**Comment by lorencinibrdev** [comment]
Hi @danrabbit , thanks for the tip on vala-lint. Worked fine here.

Could you please clarify what you meant about the PR? I checked here, like you said, seems like app_name is indeed always set on Notification class.

**Comment by danrabbit** [comment]
Sorry I meant to say that it *isnt* guaranteed that `app_name` is set.

If we have valid appinfo, we should make sure to set `app_name` based on that appinfo

**Comment by lorencinibrdev** [comment]
Think i got it now. Thanks for the help! Let me know if any change is necessary.

**Comment by lorencinibrdev** [comment]
Adding a image as evidence.

ScreenShot

**Review by danrabbit — APPROVED** [review]
Working as expected, thanks for the fix!

**Pull request merged**